com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/LaunchRequestHandler.java

```diff
@@ -166,7 +176,7 @@
             ListeningConnector listenConnector = connectors.get(0);
             Map<String, Connector.Argument> args = listenConnector.defaultArguments();
             ((Connector.IntegerArgument) args.get("timeout")).setValue(ACCEPT_TIMEOUT);
-            String address = listenConnector.startListening(args);
+            String address = context.getLaunchMode() == LaunchMode.DEBUG ? listenConnector.startListening(args) : null;
 
             String[] cmds = constructLaunchCommands(launchArguments, false, address);
             RunInTerminalRequestArguments requestArgs = null;
```

Comment:
Extract the logic about executing the launchInTerminal request to a helper method.

`CompletableFuture<void> sendLaunchInTerminalRequest(LaunchArguments launchArguments, IDebugAdapterContext context)`